### PR TITLE
e2e, persistent-ip: Add primary-UDN tests

### PR DESCRIPTION
**Pull request by RamLavi**
**What this PR does / why we need it**:
This PR adds the needed tests needed in order to cover ipam-extentions operation on VMI/VMs using primary-UDN.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Comment by RamLavi** [comment]
Added [commit](https://github.com/kubevirt/ipam-extensions/pull/71/commits/a1df094dbdcca71b5b2859c595bb7dbb07163424) to prevent the e2e from failing on restart scenario. 
@qinqon perhaps this is also needed in the secondary case?

File: test/e2e/persistentips-primary_test.go [hunk 1]
```diff
@@ -116,6 +116,8 @@ var _ = Describe("Persistent IPs on Primary UDN interface", func() {
 					WithTimeout(5 * time.Minute).
 					Should(testenv.ContainConditionVMIReady())
 
+				Expect(testenv.Client.Get(context.Background(), client.ObjectKeyFromObject(vmi), vmi)).To(Succeed())
```
Comment (oshoval) [inline test/e2e/persistentips-primary_test.go:119]: please consider making it part of the Eventually block
(also on the other places)
this what we are usually on kubevirt fwiw

Comment (RamLavi) [reply test/e2e/persistentips-primary_test.go:119]: I don't think we need eventually, because this call is right after we wait for the VMI to be ready

Comment (oshoval) [reply test/e2e/persistentips-primary_test.go:119]: you already fetch the latest vmi as part of `ThisVMI` (implicit Eventually happens for what you need)
so you can just use that value externally

func ThisVMI(vmi *kubevirtv1.VirtualMachineInstance) func() (*kubevirtv1.VirtualMachineInstance, error) {
	return func() (p *kubevirtv1.VirtualMachineInstance, err error) {
		p = &kubevirtv1.VirtualMachineInstance{}
		err = Client.Get(context.Background(), client.ObjectKeyFromObject(vmi), p)
		if apierrors.IsNotFound(err) {
			return nil, nil
		}
		return
	}
}

**Comment by oshoval** [comment]
> Added [commit](https://github.com/kubevirt/ipam-extensions/pull/71/commits/a1df094dbdcca71b5b2859c595bb7dbb07163424) to prevent the e2e from failing on restart scenario. @qinqon perhaps this is also needed in the secondary case?

for now they are stable, i wouldnt suggest to add unless we see either by code reading or by a failure that it is really needed
not sure if the exact code / flow is there (well you know better as you derived from there)

File: test/e2e/persistentips-primary_test.go (hunk 1 above)
Comment (RamLavi) [reply test/e2e/persistentips-primary_test.go:119]: true, but this change has bigger scope than what I want to achieve in this PR. 
But we should definitely should do it on a follow up PR (change ThisVMI + remove all the "fetch" that is used before the assertion)

Comment (oshoval) [reply test/e2e/persistentips-primary_test.go:119]: honestly i dont understand why to not add 
`vmi = testenv.ThisVMI(vmi)` in the first line of the Eventually and thats all

**Comment by RamLavi** [comment]
> > Added [commit](https://github.com/kubevirt/ipam-extensions/pull/71/commits/a1df094dbdcca71b5b2859c595bb7dbb07163424) to prevent the e2e from failing on restart scenario. @qinqon perhaps this is also needed in the secondary case?
> 
> for now they are stable, i wouldnt suggest to add unless we see either by code reading or by a failure that it is really needed not sure if the exact code / flow is there (well you know better as you derived from there)

ah no no ... my bad. I need to remove that commit altogether..

**Comment by RamLavi** [comment]
[Change](https://github.com/kubevirt/ipam-extensions/compare/a1df094dbdcca71b5b2859c595bb7dbb07163424..d3f56d5166d7262576c1bc997db30d8db24b8136): Rebase

File: test/e2e/persistentips-primary_test.go (hunk 1 above)
Comment (RamLavi) [reply test/e2e/persistentips-primary_test.go:119]: I intend to do so, but in a larger scope, in a follow up PR

**Comment by qinqon** [comment]
@RamLavi let's refactor current tests a little to parameterize role and VM's interface (so we can pass the proper primary UDN network binding).

**Comment by RamLavi** [comment]
closing in favor of https://github.com/kubevirt/ipam-extensions/pull/72

**Pull request closed without merging**